src/adapter.js

```diff
@@ -173,12 +173,11 @@ var createMochaReporterConstructor = function (tc, pathname) {
   }
 }
 /* eslint-disable no-unused-vars */
-var createMochaStartFn = function (mocha) {
+var createMochaStartFn = function (karma, mocha) {
   /* eslint-enable no-unused-vars */
-  return function (config) {
-    var clientArguments
-    config = config || {}
-    clientArguments = config.args
+  return function () {
+    var config = karma.config || {}
+    var clientArguments = config.args
```

Comment:
Move these two lines up before the return statement.  They will execute during file-load so the clientArguments will be available then, rather than only later during the load-event handling.

That has no impact, but in the case of karma-jasmine, the config values can be used to reset jasmine.DEFAULT_TIMEOUT_INTERVAL and it must be set during file-load.